src/java.base/share/classes/java/time/format/Parsed.java

```diff
@@ -446,6 +468,36 @@ private void resolveTimeFields() {
             updateCheckConflict(MINUTE_OF_DAY, HOUR_OF_DAY, mod / 60);
             updateCheckConflict(MINUTE_OF_DAY, MINUTE_OF_HOUR, mod % 60);
         }
+        if (dayPeriod != null) {
+            if (fieldValues.containsKey(HOUR_OF_DAY)) {
+                long hod = fieldValues.remove(HOUR_OF_DAY);
+                if (resolverStyle != ResolverStyle.LENIENT) {
+                    HOUR_OF_DAY.checkValidValue(hod);
+                }
+                updateCheckDayPeriodConflict(HOUR_OF_DAY, hod);
+            } else if (fieldValues.containsKey(HOUR_OF_AMPM)) {
+                long hoap = fieldValues.remove(HOUR_OF_AMPM);
+                if (resolverStyle != ResolverStyle.LENIENT) {
+                    HOUR_OF_AMPM.checkValidValue(hoap);
+                }
+                if (dayPeriod.includes((hoap + 12) * 60)) {
+                    hoap += 12;
+                }
+                updateCheckDayPeriodConflict(HOUR_OF_DAY, hoap);
+            } else {
+                long midpoint = dayPeriod.mid();
+                fieldValues.put(HOUR_OF_DAY, midpoint / 60);
+                fieldValues.put(MINUTE_OF_HOUR, midpoint % 60);
+                // dayPeriod precedes AmPm. Override it if exists.
+                if (fieldValues.containsKey(AMPM_OF_DAY)) {
+                    long ap = fieldValues.remove(AMPM_OF_DAY);
+                    if (resolverStyle != ResolverStyle.LENIENT) {
+                        AMPM_OF_DAY.checkValidValue(ap);
+                    }
+                    updateCheckDayPeriodConflict(AMPM_OF_DAY, midpoint / 720);
```

Comment:
No need to put `AMPM_OF_DAY` back in here because you've already resolved it to `HOUR_OF_DAY` and `MINUTE_OF_HOUR`. There probably does need to be validation to check that the day period agrees with the AM/PM value.

Reply:
Line can still be removed AFAICT

Reply:
Fixed.